packages/playwright-core/src/server/bidi/bidiFirefox.ts

```diff
@@ -77,7 +77,25 @@ export class BidiFirefox extends BrowserType {
     return env;
   }
 
-  override attemptToGracefullyCloseBrowser(transport: ConnectionTransport): void {
+  override async attemptToGracefullyCloseBrowser(transport: ConnectionTransport): Promise<void> {
+    if (!transport.onmessage) {
+      // browser.close does not work without an active session. If there is no connection
+      // created with the transport, make sure to create a new session first.
+      transport.send({ method: 'session.new', params: {
+        capabilities: {
+          alwaysMatch: {
+            unhandledPromptBehavior: { default: 'ignore' },
```

Comment:
Is there a reason why you need to set this capability? `browser.close` is closing all tabs first before shutting down the browser by not opening any beforeunload prompt.